docs/emitters/code-generation/server/server-code-generation.md

```diff
@@ -0,0 +1,375 @@
+
+3.  **Compile your TypeSpec:** Use the `tsp compile` command to generate server stubs and other artifacts based on your specification.
+
+    ```bash
+    tsp compile .
+    ```
+
+    The compiler will output the generated server stubs to the directories specified in your `tspconfig.yaml` file (e.g., `servers/aspnet/generated` for C#).
+
+4.  **Project structure:** After compiling, you will typically see the following structure:
+
+    ```text
+    \my-api
+        \tsp-output
+            \schema                  # Generated OpenAPI 3.0 spec
+                openapi.yaml
+            \clients                 # Generated Client SDK for selected language(s)
+                \dotnet
+                \node
+                ...
+            \servers                 # Generated Server Code
+                \aspnet
+                    \generated    # Generated code for ASP.NET
+                \node
+                    \generated   # Generated code for Node.js
+        main.tsp
+        tspconfig.yaml
+        package.json
+    ```
+
+## Understanding the generated code structure
+
+The server stub generation process creates a basic code structure for your server application. The specific files and folders generated depend on the target language and the complexity of your API.
+
+**C# (ASP.NET core) example:**
+
+Based on a generic 'todo' style app, the generated code structure for C# will typically include:
+
+```text
+servers/
+┣ aspnet/
+┃ ┗ generated/
+┃   ┣ controllers/               # Contains the base controller classes
+┃ ┃ ┃ ┣ CommentOpsOperationsControllerBase.cs  # Base class for comment-related API operations
+┃ ┃ ┃ ┣ CommentsOperationsControllerBase.cs   # Base class for general comment operations
+┃ ┃ ┃ ┣ ... (other controller base classes)
+┃   ┣ lib/                     # Utility functions
+┃ ┃ ┃ ┣ ArrayConstraintAttribute.cs    # Custom attributes for enforcing array constraints
+┃ ┃ ┃ ┣ Base64UrlConverter.cs        # Utility for Base64 URL encoding
+┃ ┃ ┃ ┣ ... (other utility files)
+┃   ┣ models/                  # Defines the data models (DTOs) used by the API
+┃ ┃ ┃ ┣ Attachment.cs              # Data model representing an attachment
+┃ ┃ ┃ ┣ Collaborator.cs            # Data model representing a collaborator
+┃ ┃ ┃ ┣ ... (other model files)
+┃   ┗ operations/              # Defines service interface
+┃ ┃ ┃ ┣ ICommentOpsOperations.cs  # Service interface for comment-related operations
+┃ ┃ ┃ ┣ ICommentsOperations.cs   # Service interface for general comment operations
+┃ ┃ ┃ ┣ ... (other operation interface files)
+```
+
+*   **`Controllers`**: These folders contain *base classes* for your API controllers (e.g., `CommentOpsOperationsControllerBase.cs`, `TodoItemsOperationsControllerBase.cs`). These base classes define the API endpoints and handle basic request processing. **You will need to create concrete controller classes that inherit from these base classes and implement your business logic.**
```

Comment:
This is out of date - there are no more base controllers

Reply:
revised